### PR TITLE
Add build_failure_rca validator

### PR DESCRIPTION
**Pull request by shirangj**
Signed-off-by: Shirang Jia <shiranj@amazon.com>

## What does this PR do?

Add validator for build failure RCA patterns

## How was this PR tested?

Testing using validation pipeline.

### CODE REVIEW

File: scripts/commit_validation/commit_validation/validators/build_failure_rca_validator.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    """A file-level validator that makes sure build failure RCA patterns can catch all the test cases"""
+
+    def run(self, commit: Commit, errors: List[str]) -> bool:
+        for file_name in commit.get_files():
+            for pattern in EXCLUDED_VALIDATION_PATTERNS:
+                if fnmatch.fnmatch(file_name, pattern):
+                    if VERBOSE: print(f'{file_name}::{self.__class__.__name__} SKIPPED - Validation pattern excluded on path.')
+                    break
+            else:
+                if fnmatch.fnmatch(file_name, RCA_PATTERN_PATH):
```
Comment (brianherrera) [inline scripts/commit_validation/commit_validation/validators/build_failure_rca_validator.py:28]: Should this be `if not` instead?

Comment (shirangj) [reply scripts/commit_validation/commit_validation/validators/build_failure_rca_validator.py:28]: Good catch! Not sure how I missed "not"...

File: scripts/build/build_failure_rca/rca_patterns/Compilation.json [hunk 2]
```diff
@@ -57,4 +57,5 @@
         "owners": []
       }
     ]
-  }
\ No newline at end of file
+  }
+a
```
Comment (brianherrera) [inline scripts/build/build_failure_rca/rca_patterns/Compilation.json:61]: Is this left over from a previous test?

Comment (shirangj) [reply scripts/build/build_failure_rca/rca_patterns/Compilation.json:61]: I intentionally made the wrong format to see if the validator can catch this error.
I will revert it back once the test is done

**Review by brianherrera — APPROVED** [review]

**Pull request merged**